File: gems/decomplex/examples/lua/temporal-ordering-pressure.lua
TemporalOrderExample = {}
function TemporalOrderExample:one() self.a = 1 end
function TemporalOrderExample:two() self.a = 2; self.b = 3 end
function TemporalOrderExample:three() self.b = 4 end
function TemporalOrderExample:reader() return self.a end

File: gems/decomplex/examples/kotlin/temporal-ordering-pressure.kt
class TemporalOrderExample {
  var a = 0
  var b = 0

  fun one() {
    this.a = 1
  }

  fun two() {
    this.a = 2
    this.b = 3
  }

  fun three() {
    this.b = 4
  }

  fun reader(): Int {
    return this.a
  }
}
